src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -769,13 +771,12 @@ internal bool AddTrustedLocation(string path)
         {
             try
             {
-                if (IsTrustedLocation(path) || string.IsNullOrEmpty(path))
+                PathHelper.ValidateDirectory(path);
+                if (_isTrustedLocation(path))
                 {
                     return false;
                 }
-
-                string location = PathHelper.ValidateDirectory(path);
-                trustedLocations.Add(location);
+                trustedLocations.Add(path);
```

Comment:
Can you add docs for what this function returns.

Reply:
done